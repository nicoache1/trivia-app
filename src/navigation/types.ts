import { ParamListBase } from '@react-navigation/native'
import { NativeStackScreenProps } from '@react-navigation/native-stack'
import React from 'react'

import { Routes } from './routes'

export interface ParamList extends ParamListBase {
  [Routes.Welcome]: undefined
  [Routes.ConfigureGame]: undefined
  [Routes.Game]: undefined
  [Routes.Results]: undefined
  [Routes.AnimatedSplash]: undefined
}

export type SceneProps<T extends keyof ParamList> = React.FC<
  NativeStackScreenProps<ParamList, T>
>

export type NavigationProp<T extends keyof ParamList> = NativeStackScreenProps<
  ParamList,
  T
>['navigation']
