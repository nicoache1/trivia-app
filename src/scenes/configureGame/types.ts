import { Difficulties, QuestionAmounts } from '../../types/game'

export interface ConfigureGameForm {
  name: string
  amount: number
  difficulty: number
}
