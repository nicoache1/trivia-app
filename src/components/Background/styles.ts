import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
  vector: {
    ...StyleSheet.absoluteFillObject,
    top: 70,
  },
})
