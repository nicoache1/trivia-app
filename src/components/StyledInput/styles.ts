import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  input: {
    backgroundColor: 'transparent',
  },
  labelContainer: {
    paddingHorizontal: 10,
    paddingVertical: 5,
  },
})
