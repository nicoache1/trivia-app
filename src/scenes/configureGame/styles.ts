import { Dimensions, StyleSheet } from 'react-native'

const { width } = Dimensions.get('window')

export const styles = StyleSheet.create({
  buttonContainer: {
    flex: 1,
    justifyContent: 'center',
  },
  formContainer: {
    flex: 4,
  },
  headerContainer: {
    alignItems: 'center',
    backgroundColor: 'transparent',
    justifyContent: 'center',
    width,
  },
  scene: {
    backgroundColor: 'transparent',
  },
  screenContainer: {
    flex: 1,
  },
  subtitle: {
    paddingVertical: 10,
  },
  textContainer: {
    flexDirection: 'row',
  },
  vector: {
    bottom: 0,
    left: 0,
    position: 'absolute',
    right: 0,
    top: 70,
  },
})
