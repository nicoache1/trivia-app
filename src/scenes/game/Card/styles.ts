import { Dimensions, StyleSheet } from 'react-native'

const { height } = Dimensions.get('window')

export const styles = StyleSheet.create({
  categoryContainer: {
    flex: 1,
  },
  categoryLabel: {
    borderRadius: 16,
    height: 40,
    justifyContent: 'center',
    minWidth: 120,
  },
  container: {
    borderRadius: 16,
    height: height * 0.5,
    overflow: 'hidden',
    padding: 20,
    position: 'absolute',
    width: '100%',
  },
  questionContainer: {
    flex: 2,
  },
  text: {
    textAlign: 'center',
  },
})
