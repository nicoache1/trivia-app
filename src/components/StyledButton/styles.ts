import { StyleSheet } from 'react-native'

import { ITheme } from '../../styles/types'
import { ButtonMode } from './ButtonMode'

export const textStyle = (theme: ITheme) => ({
  [ButtonMode.DARK]: {
    color: theme.colors.WHITE,
  },
  [ButtonMode.LIGHT]: {
    color: theme.colors.WHITE,
  },
})

export const buttonStyle = (theme: ITheme) => ({
  [ButtonMode.DARK]: {
    backgroundColor: theme.colors.PRIMARY,
  },
  [ButtonMode.LIGHT]: {
    backgroundColor: theme.colors.SECONDARY,
  },
})

export const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    borderRadius: 10,
    height: 56,
    justifyContent: 'center',
    paddingVertical: 10,
    width: '100%',
  },
  buttonText: {
    alignItems: 'center',
  },
})
