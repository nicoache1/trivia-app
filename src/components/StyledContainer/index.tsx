import { concat } from 'lodash'
import React from 'react'
import { StyleProp, View, ViewProps, ViewStyle } from 'react-native'

import { useTheme } from '../../styles/Theme'
import { ColorPalette } from '../../styles/types'

type CustomViewProps = {
  color?: ColorPalette
  children?: React.ReactNode
} & ViewProps

export const StyledContainer: React.FC<CustomViewProps> = props => {
  const { Theme: ActualTheme } = useTheme()

  const { color, style, children } = props
  const themedColor = color
    ? ActualTheme.colors[color]
    : ActualTheme.colors.TRANSPARENT

  const contextStyle: StyleProp<ViewStyle> = {
    backgroundColor: themedColor,
    paddingHorizontal: ActualTheme.spacing.HORIZONTAL_SCREEN_PADDING,
    paddingVertical: 10,
  }

  const mergedStyle: StyleProp<ViewStyle> = style
    ? style instanceof Array
      ? concat([contextStyle], style)
      : { ...contextStyle, ...(style as any) }
    : contextStyle

  return (
    <View {...props} style={mergedStyle}>
      {children}
    </View>
  )
}
