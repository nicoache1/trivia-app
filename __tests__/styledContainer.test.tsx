import { render } from '@testing-library/react-native'
import React from 'react'
import { StyledContainer } from 'src/components/StyledContainer'
import { ColorPalette } from 'src/styles/types'

describe('Test styledContainer', () => {
  test('Expect StyledContainer to be rendered correctly with normal color', async () => {
    const renderedSnapshot = render(<StyledContainer />).toJSON()

    expect(renderedSnapshot).toMatchSnapshot()
  })

  test('Expect StyledContainer to be rendered with some color', async () => {
    const renderedSnapshot = render(
      <StyledContainer color={ColorPalette.BLUE_SURFACE} />,
    ).toJSON()

    expect(renderedSnapshot).toMatchSnapshot()
  })
})
