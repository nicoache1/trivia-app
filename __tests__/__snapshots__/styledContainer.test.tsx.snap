// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test styledContainer Expect StyledContainer to be rendered correctly with normal color 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
/>
`;

exports[`Test styledContainer Expect StyledContainer to be rendered with some color 1`] = `
<View
  color="BLUE_SURFACE"
  style={
    Object {
      "backgroundColor": "#98D0E8",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
/>
`;
